CRM/Contribute/BAO/Contribution.php

```diff
@@ -4721,15 +4721,31 @@ public static function sendMail(&$input, &$ids, $contributionID, &$values, $recu
       throw new CRM_Core_Exception('Contribution does not exist');
     }
     $contribution->loadRelatedObjects($input, $ids, TRUE);
-    // set receipt from e-mail and name in value
     if (!$returnMessageText) {
-      $session = CRM_Core_Session::singleton();
-      $userID = $session->get('userID');
-      if (!empty($userID)) {
-        list($userName, $userEmail) = CRM_Contact_BAO_Contact_Location::getEmailDetails($userID);
-        $values['receipt_from_email'] = CRM_Utils_Array::value('receipt_from_email', $input, $userEmail);
-        $values['receipt_from_name'] = CRM_Utils_Array::value('receipt_from_name', $input, $userName);
+      // Set receipt from e-mail and name in value if we are sending the message.
+      $userEmail = NULL;
+      $userName = NULL;
+
+      // First, try to retrieve default name and email address.
+      $sitewide_default_from = array_pop(CRM_Core_OptionGroup::values('from_email_address', NULL, NULL, NULL, ' AND is_default = 1'));
```

Comment:
@jmcclelland Jamie why not use the domain from email here rather than the default from the list of from emails?